File: gamerental/src/main/java/com/example/app/user/GetUsers.java
package com.example.app.user;

import io.fluxzero.sdk.Fluxzero;
import io.fluxzero.sdk.tracking.handling.HandleQuery;
import io.fluxzero.sdk.tracking.handling.Request;

import java.util.List;

public record GetUsers() implements Request<List<UserProfile>> {
    @HandleQuery
    List<UserProfile> handle() {
        return Fluxzero.search(UserProfile.class).fetchAll();
    }
}

File: flux-basic-kotlin/src/main/kotlin/com/example/app/user/api/GetUsers.kt
package com.example.app.user.api

import com.example.app.user.api.model.UserProfile
import io.fluxzero.sdk.Fluxzero
import io.fluxzero.sdk.common.serialization.FilterContent
import io.fluxzero.sdk.tracking.handling.HandleQuery
import io.fluxzero.sdk.tracking.handling.Request

data class GetUsers(val dummy: Unit = Unit) : Request<List<UserProfile>> {
    @HandleQuery
    @FilterContent
    fun handle(): List<UserProfile> {
        return Fluxzero.search(UserProfile::class.java).fetchAll()
    }
}
